pkg/kube/inject/webhook.go

```diff
@@ -1163,3 +1243,50 @@ func handleError(message string) {
 	log.Errorf(message)
 	totalFailedInjections.Increment()
 }
+
+// modifies the pod to ensure that the proxy runs with the given proxyUID instead of 1337
+func replaceProxyRunAsUserID(pod *corev1.Pod, proxyUID *int64) {
```

Comment:
I am somewhat uncomfortable with this. This is both extremely coupled the injection template logic, and we have been trying to move logic into the templates instead of post-processing the pod where possible.

however, I think I need answers to some of my other comments to make any recommendations on alternatives